### PR TITLE
Progress bar

### PR DESCRIPTION
**Pull request by qiyundai**
Changes made to index page:
- Each card has a progress bar now.
- The progress bar only starts tracking if each item has at least one duty associated. Otherwise, a warning bar appears.
screencapture-localhost-3000-2021-04-16-12_12_17

Changes made to show page to accommodate progress bar:
- a "not needed" is added to each item's dropdown manu to remove the associated duty that's no longer relevant.

screencapture-localhost-3000-trips-14-2021-04-16-12_12_52

### CODE REVIEW

File: app/views/trips/card/_trip_card.html.erb [hunk 1]
```diff
@@ -38,7 +38,19 @@
             
           <% end %>
         </div>
-
+        <% @duties_done = trip.checklist.duties.count { |duty| duty.status == "✅" } %>
+        <% @duties_all = trip.checklist.duties.count %>
+        <% if trip.checklist_items.count { |item| item.duties.first != nil } == trip.checklist_items.count %>
+          <div class="progress">
+            <div class="progress-bar" role="progressbar" style="width: <%= (@duties_done/@duties_all.to_f*100).round %>%" aria-valuenow="<%= (@duties_done/@duties_all.to_f*100).round %>" aria-valuemin="0" aria-valuemax="100">
```
Comment (fredThem) [inline app/views/trips/card/_trip_card.html.erb:45]: Line is breaking the app for new users
Screen Shot 2021-04-16 at 8 15 31 PM

**Review by fredThem — CHANGES_REQUESTED** [review]
Let’s test with two browser. One with Jane and a new user with no trips. It’s always with a new user scenario that I get the app to crash**

It works for a logged in user * and looks very good !!!

**Comment by fredThem** [comment]
Not sure about the need at least one duty for each item to*
What if a new item is added the progress is removed :(
I see your logic here. With the duty asssign this seem more complicated than I expected 😩 
Let's talk about it.

**Comment by qiyundai** [comment]
I can see this crashing with no trips associated to the user. I'll create a user with no trip and debug through it later.

**Comment by qiyundai** [comment]
> Not sure about the need at least one duty for each item to*
> What if a new item is added the progress is removed :(
> I see your logic here. With the duty asssign this seem more complicated than I expected 😩
> Let's talk about it.

I thought about this for a long time. Eventually, I thought it's still better if the progress bar gets yellowed out with the 'at least one duty' message to remind people they have unassigned items in there checklist than to have a progress bar that starts tracking regardless of the duties and gives a very inaccurate reflection of the actual progress.

If you've thought of any other way to go about this, please let me know and I'll try to implement it differently.

**Comment by fredThem** [comment]
Screen Shot 2021-04-17 at 8 29 51 AM


@qiyundai  Still breaks for new users :S with one empty trip on home page.

**Comment by qiyundai** [comment]
That's right. I didn't think of the case scenario where the user created a trip without a single activity. Fix coming right up.

**Comment by qiyundai** [comment]
screencapture-localhost-3000-2021-04-17-08_42_42
the reason it was breaking is that the progress bar tries to render progress regardless if the checklist is empty or not. Now it knows better.

**Pull request merged**